src/services/Api.ts

```diff
@@ -59,24 +60,31 @@ const newRequest = <P>(method: HTTP_VERBS, url: string, queryParams: any, data:
     params: queryParams
   });
 
+interface LoginRequest {
+  username: string;
```

Comment:
`string` type is very generic -- it is just string after all. Let's use `UserName` so we can search the codebase for this and have type safety.